app/src/main/java/com/github/factotum_sdp/factotum/data/localisation/Location.kt

```diff
@@ -18,11 +19,49 @@ import java.util.concurrent.CountDownLatch
  */
 class Location(query: String, context : Context) {
 
+    companion object {
+        const val CACHE_FILE_NAME = "locations.txt"
+        const val CACHE_FILE_SEPARATOR = "|"
+
+        /**
+         * Creates a location and stores it in a file named CACHE_FILE_NAME in cache if an address was found.
+         * Stored as : addressName CACHE_FILE_SEPARATOR latitude CACHE_FILE_SEPARATOR longitude
+         *
+         * @param query : String. Location to add
+         * @param context : Context. Context in which this method is called
+         * @return returns the location that has been added to the file, or null if no address was found
+         */
+        fun createAndStore(query: String, context: Context): Location?{
+            val location = Location(query, context)
+            if(location.address == null){
+                return location
```

Comment:
For me this does not return null if no address was found rather a Location object with a null address. If that's what you wanted maybe updated your description, otherwise return null so it's a bit clearer.

Reply:
Good comment thank you !